Compile warnings when building Oboe with AAudio

Build warnings when building AAudio with `mmm frameworks/av/media/libaaudio`

[ 72% 34/47 9s remaining] //frameworks/av/media/libaaudio/examples/loopback:aaudio_loopback clang++ sr
In file included from frameworks/av/media/libaaudio/examples/loopback/src/loopback.cpp:42:
external/oboe/apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h:78:33: warning: comparison of
 floating-point type 'double' with enumeration type 'GlitchAnalyzer::constants' is deprecated [-Wdepre
cated-enum-float-conversion]
   78 |             if (signalToNoiseDB < MIN_SNR_DB) {
      |                 ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~
frameworks/av/media/libaaudio/examples/loopback/src/loopback.cpp:439:27: warning: variable 'gain' set 
but not used [-Wunused-but-set-variable]
  439 |     double                gain                       = 1.0;
      |                           ^
2 warnings generated.
[ 74% 35/47 9s remaining] //frameworks/av/media/libaaudio/examples/loopback:aaudio_loopback clang++ sr
In file included from frameworks/av/media/libaaudio/examples/loopback/src/loopback.cpp:42:
external/oboe/apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h:78:33: warning: comparison of
 floating-point type 'double' with enumeration type 'GlitchAnalyzer::constants' is deprecated [-Wdepre
cated-enum-float-conversion]
   78 |             if (signalToNoiseDB < MIN_SNR_DB) {
      |                 ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~
frameworks/av/media/libaaudio/examples/loopback/src/loopback.cpp:439:27: warning: variable 'gain' set 
but not used [-Wunused-but-set-variable]
  439 |     double                gain                       = 1.0;
      |                           ^
2 warnings generated.

RBE Stats: down 0 B, up 0 B, 2 racing locals

--- a/apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h
+++ b/apps/OboeTester/app/src/main/cpp/analyzer/GlitchAnalyzer.h
@@ -75,7 +75,7 @@ public:
         } else {
             double signalToNoise = mMeanSquareSignal / mMeanSquareNoise; // power ratio
             double signalToNoiseDB = 10.0 * log(signalToNoise);
-            if (signalToNoiseDB < MIN_SNR_DB) {
+            if (signalToNoiseDB < static_cast<float>(MIN_SNR_DB)) {
                 setResult(ERROR_VOLUME_TOO_LOW);
             }
             return signalToNoiseDB;

--- a/src/opensles/AudioStreamOpenSLES.cpp
+++ b/src/opensles/AudioStreamOpenSLES.cpp
@@ -72,8 +72,8 @@ Result AudioStreamOpenSLES::open() {
 
     // OpenSL ES only supports I16 and Float
     if (mFormat != AudioFormat::I16 && mFormat != AudioFormat::Float) {
-        LOGW("%s() Android's OpenSL ES implementation only supports I16 and Float. Format: %d",
-             __func__, mFormat);
+        LOGW("%s() Android's OpenSL ES implementation only supports I16 and Float. Format: %s",
+             __func__, oboe::convertToText(mFormat));
         return Result::ErrorInvalidFormat;
     }
 
@@ -422,7 +422,7 @@ bool AudioStreamOpenSLES::processBufferCallback(SLAndroidSimpleBufferQueueItf bq
         LOGD("Oboe callback returned Stop");
         shouldStopStream = true;
     } else {
-        LOGW("Oboe callback returned unexpected value = %d", result);
+        LOGW("Oboe callback returned unexpected value = %d", static_cast<int>(result));
         shouldStopStream = true;
     }
     if (shouldStopStream) {

